v1/events.ts

```diff
@@ -0,0 +1,64 @@
+type TODO = unknown;
+
+// Common API client contract
+interface ApiClient {
+  fetchJson(opts: {
+    method: "GET" | "POST";
+    path: string;
+    query?: URLSearchParams;
+    body?: unknown;
+  }): Promise<unknown>;
+}
+
+export default class DatadogEventsApi {
+  #api: ApiClient;
+  constructor(api: ApiClient) {
+    this.#api = api;
+  }
+
+  async submit(data: Event): Promise<string> {
+    const json = await this.#api.fetchJson({
+      path: `/api/v1/events`,
+      method: "POST",
+      body: {
+        text: data.text,
+        title: data.title,
+        aggregation_key: data.aggregation_key,
+        alert_type: data.alert_type,
+        date_happened: data.date_happened,
+        device_name: data.device_name,
+        host: data.host,
+        priority: data.priority,
+        related_event_id: data.related_event_id,
+        source_type_name: data.source_type_name,
+        tags: data.tags ?? [],
+      },
+    });
+    return (json as { status: string }).status;
+  }
+}
+
+export interface Event {
+  /** max length: 4000 **/
+  text: string;
+  /** max length: 100 **/
+  title: string;
+  aggregation_key?: string;
+  alert_type?: AlertType;
+  date_happened?: number;
```

Comment:
To reduce programming errors, let's accept this as a Date and then transform to a number on-the-spot, like the service checks API: https://github.com/cloudydeno/deno-datadog_api/blob/c8f07a8b154774410857ae69d5fb81c9f4c2e58a/v1/service_checks.ts#L29

Reply:
Changes implemented!